Sources/RestKit/RestError.swift

```diff
@@ -17,32 +17,63 @@
 import Foundation
 
 /// An error from processing a network request or response.
-public enum RestError: Error {
+public enum RestError {
 
     /// No response was received from the server.
     case noResponse
 
     /// No data was returned from the server.
     case noData
 
+    /// Failed to save the downloaded data.
+    /// The specified file may already exist or the disk may be full.
+    case saveData
+
     /// Failed to serialize value(s) to data.
-    case serializationError
+    case serialization
 
     /// Failed to replace special characters in the
     /// URL path with percent encoded characters.
-    case encodingError
-
-    /// FileManager failed to handle the given file.
-    /// (The file may already exist or the disk may be full.)
-    case fileManagerError
-
-    /// Failed to load the given file.
-    case invalidFile
+    case encoding
 
     /// The request failed because the URL was malformed.
     case badURL
 
-    /// An HTTP error with a status code and description.
-    case failure(Int, String)
+    /// Generic HTTP error with a status code and description.
+    case http(statusCode: Int?, message: String?)
+
+}
```

Comment:
Is there any way to add an optional Dictionary like `userInfo` of NSError?  There are some cases where additional error information is available and it would be nice to have some way to pass it back in the error object.

Reply:
Ah, I forgot that some of the services have multiple entries in their error JSON. 

Discovery, NLC, and NLUE have both an "error" and a "description" field.
PersonalityInsights has both an "error" and a "help" field.
SST and TTS have an "error" and a "code_description" field.
VR has a "status", "statusInfo", "error", "description", "error_id", and "error" field.

I think your suggestion would work well. I'm debating between keeping the `message` string and adding an additional `metadata` field, or removing `message` and putting everything in `metadata`. What do you think?

Reply:
I think we could put everything in `metadata` as long as the `errorDescription` returns `metadata["message"]` for http errors.